### PR TITLE
Expose endpoint for searching for person using ssn and last name.

### PR DESCRIPTION
**Pull request by bjorntore**
The idea behind this new endpoint is that it should be used by the component in frontend used for looking up people that should be signees.

1. Can we expose person search to the frontend like this? ⚠️ Or should we just confirm the SSNs and last names on the backend before sending correspondence and delegating access? @RonnyB71 
2. Do we just return the full Person object as a response, or have I taken the correct approach using a DTO with a subset of data?

### CODE REVIEW

**Comment by HauklandJ** [comment]
Returning a minimized personDTO is imo a better approach than returning the full object. This way we keep the response to only that which is needed to populate the field.

**Review by martinothamar — COMMENTED** [review]
+1 for using a dedicated type for the API contract

* When do we use `class` and when do we use `struct`? 
* `Dto`-suffix vs `Request`/`Response`?

**Comment by bjorntore** [comment]
> +1 for using a dedicated type for the API contract
> 
>     * When do we use `class` and when do we use `struct`?
> 
>     * `Dto`-suffix vs `Request`/`Response`?

Yep, want answers to these questions. :P

File: src/Altinn.App.Api/Controllers/PersonSearchController.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        CancellationToken cancellationToken
+    )
+    {
+        Person? person = await _personClient.GetPerson(
+            personSearchRequest.SocialSecurityNumber,
+            personSearchRequest.LastName,
+            cancellationToken
+        );
+
+        return person is null ? NotFound() : PersonSearchResponse.MapFromPerson(person);
```
Comment (ivarne) [inline src/Altinn.App.Api/Controllers/PersonSearchController.cs]: Is 404 the correct status code when a search returns no results? Isn't it better to have 200 and modify `PersonSearchResponse` to have a `Success` property. The code is mostly for looking at logs, and a miss is a valid result and no reason to investigate.

Somehow Swagger seems to think that you return `ProblemDetails` for the 404, so if we keep the code, it need to actually return what we promised.

Comment (martinothamar) [reply src/Altinn.App.Api/Controllers/PersonSearchController.cs]: alternatively 204 No Content

Comment (bjorntore) [reply src/Altinn.App.Api/Controllers/PersonSearchController.cs]: @ivarne Take another look.

**Review by HauklandJ — APPROVED** [review]

**Pull request merged**